### ISSUE TITLE
'isDistributedPreviousStatement' return wrong result when previous line is a comment

### ISSUE THREAD

**Issue by Gogomoe** [checkstyle/checkstyle#9925]
For the following code
```java
    void foo() {
        List.of(1, 2, 3).stream()
                .map(x -> x * x)
                .filter(x -> x % 2 != 0)
                // comment 1
                .collect(Collectors.toList());
        // comment 2
    }
```
`CommentsIndentationCheck.isDistributedPreviousStatement(comment2)` should return true but returns false.

In some cases, this will produce false warning.
```plain
[WARN] ~\src\Test.java:11:9: Comment has incorrect indentation level 8, expected is 16
```

The reason for this bug is similar to #9105 . 
`isDistributedExpression`  try to find the first token of the statement.
It should find `List`, but the comment1 breaks the flow and it returns the comment.

I think I can fix it.

**Comment by rnveach** [comment]
Issue is not present on 9.0:
````
$ cat TestClass.java
public class TestClass {
    void foo() {
        List.of(1, 2, 3).stream()
                .map(x -> x * x)
                .filter(x -> x % 2 != 0)
                // comment 1
                .collect(Collectors.toList());
        // comment 2
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="CommentsIndentationCheck" />
    </module>
</module>

$ java -jar checkstyle-9.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````


Issue is present on 8.43:
````
$ cat TestClass.java
public class TestClass {
    void foo() {
        List.of(1, 2, 3).stream()
                .map(x -> x * x)
                .filter(x -> x % 2 != 0)
                // comment 1
                .collect(Collectors.toList());
        // comment 2
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="CommentsIndentationCheck" />
    </module>
</module>

$ java -jar checkstyle-8.43-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:8:9: Comment has incorrect indentation level 8, expected is 16, indentation should be the same level as line 7. [CommentsIndentation]
Audit done.
Checkstyle ends with 1 errors.
````

**Comment by rnveach** [comment]
I believe this was fixed as part of #9105 and so I am closing it.
